### PR TITLE
WT-7520 Add start and stop values to automatic flag generation code

### PR DESCRIPTION
**Pull request by jiechenbo**
This ticket involves adding the start and stop values to the automatic flag generation code. I found the best way to represent the start and stop is through setting the values of start on the start generation line, and values of stop on the stop generation line. I have then done further checks to see if there are more flags than the stop limit, and to start of an offest from start.

### CODE REVIEW

File: dist/flags.py [hunk 1]
```diff
@@ -18,30 +19,46 @@ def flag_declare(name):
...
                     sys.exit(1)
 
                 # Calculate number of hex bytes, create format string
-                fmt = "0x%%0%dxu" % ((len(defines) + 3) / 4)
+                fmt = "0x%%0%dxu" % ((start + len(defines) + 3) / 4)
 
+                # Generate the flags starting from an offset set from the start value.
                 tfile.write(header)
-                v = 1
+                v = 1 << start
```
Comment (tetsuo-cpp) [inline dist/flags.py:54]: 🧠

File: dist/flags.py [hunk 2]
```diff
@@ -18,30 +19,46 @@ def flag_declare(name):
 
         lcnt = 0
         parsing = False
+        start = 0
         for line in f:
             lcnt = lcnt + 1
             if line.find('AUTOMATIC FLAG VALUE GENERATION START') != -1:
+                m = re.search("\d+", line)
+                if m == None:
+                    print(name + ": automatic flag generation start line " + str(lcnt) +\
```
Comment (tetsuo-cpp) [inline dist/flags.py:28]: Maybe include example in error msg.

**Comment by sueloverso** [comment]
@keithbostic I added you as a reviewer as since you're the one who originally added the START/STOP flags.

**Review by sueloverso — APPROVED** [review]
LGTM. How did you test this?

**Review by keithbostic — APPROVED** [review]
lgtm. I have a mild preference for making the default start value 0 so you don't have to list it everywhere, but shrug, it's fine with me either way.

**Comment by jiechenbo** [comment]
>How did you test this? 
I have tested this feature through:
1.  Modifying the start value multiple times and looking if the offsets worked
2. Modifying the stop value multiple times and looking if the flags.py would detect the boundaries
3. Modified both start and stop, to distinguish the number of flags allowed from start to end boundary.

@keithbostic , personally I find that removing the 0, to a default value might be a little confusing, since there will be some places having numbers in the flag generation comment, and no number in the flag generation comment

File: dist/flags.py (hunk 2 above)
Comment (jiechenbo) [reply dist/flags.py]: done!

File: src/include/cell.h [hunk 3]
```diff
@@ -144,10 +144,10 @@ struct __wt_cell {
     uint8_t __chunk[1 + 1 + 1 + 7 * WT_INTPACK64_MAXSIZE + WT_INTPACK32_MAXSIZE];
 };
 
-/* AUTOMATIC FLAG VALUE GENERATION START */
+/* AUTOMATIC FLAG VALUE GENERATION START 0 */
 #define WT_CELL_UNPACK_OVERFLOW 0x1u            /* cell is an overflow */
 #define WT_CELL_UNPACK_TIME_WINDOW_CLEARED 0x2u /* time window cleared because of restart */
-                                                /* AUTOMATIC FLAG VALUE GENERATION STOP */
+/*AUTOMATIC FLAG VALUE GENERATION STOP 8 */
```
Comment (tetsuo-cpp) [inline src/include/cell.h:150]: Did this get generated properly? A space got removed.

**Review by tetsuo-cpp — APPROVED** [review]
LGTM aside from that formatting oddity. Nice work @jiechenbo.

File: src/include/cell.h (hunk 3 above)
Comment (jiechenbo) [reply src/include/cell.h]: thanks! I accidentally removed the space, when verifying the changes.

**Pull request merged**